avm/ptn/sa/conversation-knowledge-mining/main.bicep

```diff
@@ -0,0 +1,1168 @@
+    64
+  )
+  properties: {
+    mode: 'Incremental'
+    template: {
+      '$schema': 'https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#'
+      contentVersion: '1.0.0.0'
+      resources: []
+      outputs: {
+        telemetry: {
+          type: 'String'
+          value: 'For more information, see https://aka.ms/avm/TelemetryInfo'
+        }
+      }
+    }
+  }
+}
+
+module avmManagedIdentity 'br/public:avm/res/managed-identity/user-assigned-identity:0.4.0' = {
+  name: format(avmDeploymentNameFormat, varManagedIdentityResourceName)
+  params: {
+    name: varManagedIdentityResourceName
+    tags: tags
+    location: managedIdentityLocation
+    enableTelemetry: enableTelemetry
+  }
+}
+
+// NOTE: This assignment should leverage AVM module [avm/res/resources/resource-group](https://azure.github.io/Azure-Verified-Modules/indexes/bicep/bicep-resource-modules/), but current target scope is resourceGroup
+// TODO: Owner permissions to RG is an unsecure practice, fine grain permissions
+module assignResourceGroupOwner 'modules/rbac-rg-owner.bicep' = {
+  name: format(localModuleDeploymentNameFormat, 'rbac-rg-owner')
+  params: {
+    managedIdentityResourceId: avmManagedIdentity.outputs.resourceId
+    managedIdentityPrincipalId: avmManagedIdentity.outputs.principalId
+  }
+}
+
+// ========== Log Analytics Workspace ========== //
+module avmLogAnalyticsWorkspace 'br/public:avm/res/operational-insights/workspace:0.11.1' = {
+  name: format(avmDeploymentNameFormat, varLogAnalyticsWorkspaceResourceName)
+  params: {
+    name: varLogAnalyticsWorkspaceResourceName
+    tags: tags
+    location: logAnalyticsWorkspaceLocation
+    enableTelemetry: enableTelemetry
+    skuName: logAnalyticsWorkspaceSkuName
+    dataRetention: logAnalyticsWorkspaceDataRetentionInDays
+  }
+}
+
+// ========== Key Vault ========== //
+module avmKeyVault 'br/public:avm/res/key-vault/vault:0.12.1' = {
+  name: format(avmDeploymentNameFormat, varKeyVaultResourceName)
+  params: {
+    name: varKeyVaultResourceName
+    tags: tags
+    location: keyVaultLocation
+    enableTelemetry: enableTelemetry
+    diagnosticSettings: [{ workspaceResourceId: avmLogAnalyticsWorkspace.outputs.resourceId }]
+    sku: keyVaultSku
+    createMode: keyVaultCreateMode
+    enableVaultForDeployment: true
+    enableVaultForDiskEncryption: true
+    enableVaultForTemplateDeployment: true
+    enableSoftDelete: keyVaultSoftDeleteEnabled
+    softDeleteRetentionInDays: keyVaultSoftDeleteRetentionInDays
+    enablePurgeProtection: keyVaultPurgeProtectionEnabled
+    publicNetworkAccess: 'Enabled'
+    enableRbacAuthorization: true
+    roleAssignments: concat(
+      [
+        {
+          principalId: avmManagedIdentity.outputs.principalId
+          principalType: 'ServicePrincipal'
+          roleDefinitionIdOrName: '00482a5a-887f-4fb3-b363-3b7fe8e74483' //NOTE: Built-in role 'Key Vault Administrator'
```

Comment:
I think you can even use the name here as the Key Vault module should have its ID backed in:
```bicep
roleDefinitionIdOrName: 'Key Vault Administrator'
```

Reply:
The same is true for other resources. I guess that's why you added the `NOTE:` in the back?